syntax = "proto3";

package subscription;

service SubscriptionService {
  rpc CreateSubscription (CreateSubscriptionRequest) returns (CreateSubscriptionResponse);
  rpc GetSubscriptions (GetSubscriptionsRequest) returns (GetSubscriptionsResponse);
  rpc ChangeSubscription (ChangeSubscriptionRequest) returns (ChangeSubscriptionResponse);
  rpc DeleteSubscription (DeleteSubscriptionRequest) returns (DeleteSubscriptionResponse);
  rpc ActivateSubscription (ActivateSubscriptionRequest) returns (ActivateSubscriptionResponse);
  rpc DeactivateSubscription (DeactivateSubscriptionRequest) returns (DeactivateSubscriptionResponse);
}

// Create
message CreateSubscriptionRequest {
  string email = 1;
  string subscription_type = 2;
}

message CreateSubscriptionResponse {
  string message = 1;
}

// Read
message GetSubscriptionsRequest {}
message GetSubscriptionsResponse {
  repeated Subscription subscriptions = 1;
}

// Update
message ChangeSubscriptionRequest {
  string email = 1;
  string subscription_type = 2;
}
message ChangeSubscriptionResponse {
  string message = 1;
}

// Delete
message DeleteSubscriptionRequest {
  string email = 1;
}
message DeleteSubscriptionResponse {
  string message = 1;
}

// Activate
message ActivateSubscriptionRequest {
  string email = 1;
}
message ActivateSubscriptionResponse {
  string message = 1;
}

// Deactivate
message DeactivateSubscriptionRequest {
  string email = 1;
}
message DeactivateSubscriptionResponse {
  string message = 1;
}

message Subscription {
  string email = 1;
  string subscription_type = 2;
  bool is_active = 3;
}
